lib/clustering/mean_shift.cpp

```diff
@@ -0,0 +1,44 @@
+/// \file
+/// Maintainer: Luzian Hug
+/// Created: 25.03.2018
+///
+///
+
+
+#include "mean_shift.h"
+#include <stdlib.h>
+#include <time.h>
+namespace MouseTrack {
+
+MeanShift::MeanShift(int k) : _k(k) {
+    //empty
+}
+
+std::vector<Cluster> MeanShift::operator()(const PointCloud& cloud) const {
+
+	std::vector<Cluster> clustering = std::vector<Cluster>(_k);
+
+  //Initialize RNG
+	srand(time(NULL));
```

Comment:
It really depends, what you want to use it for. If you generate passwords, this is a very good idea.
If you write physical simulations or run an algorithm, you want a more control/a fixed seed (choose your favorite integer). 

It's all about repeatability: If you use something "random" like time, debugging might get really tricky, since you don't know the seed that causes a certain bug. This is desired in cryptographic applications (like password encryption or so). For nearly everything else, having this non-deterministic component, can cause a lot of problems.